boot_services/src/boot_services.rs

```diff
@@ -1111,6 +1180,99 @@ impl BootServices for StandardBootServices<'_> {
         }
     }
 
+    fn load_image(
+        &self,
+        boot_policy: bool,
+        parent_image_handle: efi::Handle,
+        device_path: *mut efi::protocols::device_path::Protocol,
+        source_buffer: Option<&[u8]>,
+    ) -> Result<efi::Handle, efi::Status> {
+        let load_image = self.efi_boot_services().load_image;
+        if load_image as usize == 0 {
+            panic!("function not initialize.");
+        }
+        let source_buffer_ptr =
+            source_buffer.map_or(ptr::null_mut(), |buffer| buffer.as_ptr() as *const _ as *mut c_void);
+        let source_buffer_size = source_buffer.map_or(0, |buffer| buffer.len());
+        let mut image_handle = MaybeUninit::uninit();
+        match load_image(
+            boot_policy.into(),
+            parent_image_handle,
+            device_path,
+            source_buffer_ptr,
+            source_buffer_size,
+            image_handle.as_mut_ptr(),
+        ) {
+            s if s.is_error() => Err(s),
+            _ => Ok(unsafe { image_handle.assume_init() }),
+        }
+    }
+
+    fn start_image<'a>(
+        &'a self,
+        image_handle: efi::Handle,
+    ) -> Result<(), (efi::Status, Option<BootServicesBox<'a, [u8], Self>>)> {
+        let start_image = self.efi_boot_services().start_image;
+        if start_image as usize == 0 {
+            panic!("function not initialize.");
+        }
+        let mut exit_data_size = MaybeUninit::uninit();
+        let mut exit_data = MaybeUninit::uninit();
+        match start_image(image_handle, exit_data_size.as_mut_ptr(), exit_data.as_mut_ptr()) {
+            s if s.is_error() => {
+                let data = (exit_data.as_ptr() == ptr::null()).then(|| unsafe {

```

Comment:
Is the conditional right? I think it should only process exit data if it is not null, right?
```suggestion
                let data = (!exit_data.as_ptr().is_null()).then(|| unsafe {

```